File: 424-longest-repeating-character-replacement/longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        unordered_map<int,int> check;
        int ans=0;
        int left =0, right = 0, maxfreq = 0;
        while(right<s.size())
        {
            check[s[right]-'A']++;
            maxfreq= max(maxfreq,check[s[right]-'A']);
            cout<<"r l m "<<right<<" "<<left<<" "<<maxfreq<<endl;
            if((right - left+1)-maxfreq>k){
                check[s[left]-'A']--;
                left++;
            }
              ans = max(ans,right+1-left);
              right++;
              cout<<"maxlen"<<ans<<endl;

        }
        return ans;
        
    }
};

File: 424-longest-repeating-character-replacement/longest-repeating-character-replacement.java
class Solution {
    public int characterReplacement(String s, int k) {
        int[] freq = new int[26];
        int maxfreq = 0;
        int left = 0;
        int right = 0;
        int maxLen = 0;
        while(right<s.length())
        {
            freq[s.charAt(right)-'A']++;
            maxfreq = Math.max(maxfreq,freq[s.charAt(right)-'A']);

            while(right-left+1>maxfreq + k)
            {
                freq[s.charAt(left)-'A']--;
                left++;
            }
            maxLen = Math.max(right-left+1,maxLen);
            right++;
        }
        return maxLen;
    }
}